src/expand.cpp

```diff
@@ -86,6 +87,7 @@ namespace Sass {
 
   Statement* Expand::operator()(Ruleset* r)
   {
+    bool old_at_root_without_rule = this->at_root_without_rule;
```

Comment:
Unfortunately this doesn't work in all cases like here when I need to pop off that stack at a specific point. Waiting until the end of the function would alter the implementation. IMO we should aim to use macro where possible and assume if we're not it's intentional.